src/compat/stream.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  Observable,
+  ReplaySubject,
+} from "rxjs";
+import {
+  filter,
+  map,
+  multicast,
+  refCount
+} from "rxjs/operators";
+import { isSamsungInternet } from "./constants";
+import { onPlayPause$ } from "./events";
+
+/**
+ * onPlayPause$ wrapper.
+ * Has a specific behavior when playback is happening on android Samsung internet.
+ * @param {HTMLMediaElement} mediaElement
+ * @param {boolean} autoPlay
+ * @param {Object|undefined} options
+ * @returns {Observable}
+ */
+export function onPlayPause(
+  mediaElement : HTMLMediaElement,
+  autoPlay : boolean,
+  onPlayPauseOptions? : {
+    hasLoadedMetadata: boolean;
+    initialMediaDuration : number;
+  }
+): Observable<boolean> {
+  return onPlayPause$(mediaElement).pipe(
+    filter((x, i) => {
+      if (
+        onPlayPauseOptions &&
+        onPlayPauseOptions.initialMediaDuration === 0 &&
+        onPlayPauseOptions.hasLoadedMetadata &&
+        isSamsungInternet
+      ) {
+        // On samsung, we play a first time, even when autoPlay is false.
+        // We should filter on it
+        return !(i === 0 && !autoPlay && x.type === "play");
+      }
+      return true;
+    }),
+    map((x) => (x && x.type === "play") ? true : false),
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Simplify this expression. ([typescript:S1125](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS1125))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=385&issues=AWc7CuF3iCTqmtbHx9Qn&open=AWc7CuF3iCTqmtbHx9Qn)

[ISSUE_KEY]: AWc7CuF3iCTqmtbHx9Qn